src/Smearing.jl

```diff
@@ -23,6 +23,31 @@ Derivative of the occupation function, approximation to minus the delta function
 """
 occupation_derivative(S::SmearingFunction, x) = ForwardDiff.derivative(x -> occupation(S, x), x)
 
+"""
+(f(x) - f(y))/(x - y), computed stably in the case where x and y are close
+"""
+function occupation_divided_difference(S::SmearingFunction, x, y, εF, T)
+    if T == 0
+        if x == y
+            zero(x)
+        else
+            fx = x < εF ? 1 : 0
+            fy = y < εF ? 1 : 0
+            (fx-fy)/(x-y)
+        end
+    else
+        f(z) = occupation(S, (z-εF)/T)
+        fp(z) = occupation_derivative(S, (z-εF)/T)/T
+        divided_difference_(f, fp, x, y)
+    end
+end
+function divided_difference_(f, fp, x, y)
```

Comment:
It's an internal function, I thought it made more sense to have it separate for readability (esp. if we make it more accurate in the future). Since it's internal (as should be denoted by the underscore, or do we put it in front?) it makes sense to have no free space

Reply:
I use "p" for prime consistently-ish, I don't have any better idea (prime is too verbose)

Reply:
Yes but sometimes it means derivative and sometimes just "another" so if you can distinguish that, it would be good, I guess.